include/can/firmware/hal_can_bus.hpp

```diff
@@ -0,0 +1,58 @@
+#pragma once
+
+#include <array>
+
+#include "can/core/can_bus.hpp"
+#include "can/core/ids.hpp"
+#include "can/core/parse.hpp"
+#include "stm32g4xx_hal_fdcan.h"
+
+using namespace can_bus;
+using namespace can_ids;
+
+/**
+ * HAL FD CAN wrapper. Matches the CanBus concept.
+ */
+class HalCanBus {
+  public:
+    /**
+     * Construct
+     * @param handle A pointer to an initialized FDCAN_HandleTypeDef
+     */
+    explicit HalCanBus(FDCAN_HandleTypeDef* handle);
```

Comment:
by forward declaring `FDCAN_HandleTypeDef` you could avoid having to include the hal_fdcan here. You'd lose the ability to handle HAL_StatusTypeDef but whatever

Reply:
But I'm also using `FDCAN_TxHeaderTypeDef` in this class. I can't forward declare that.